pkg/env-tests/scheduler_test.go

```diff
@@ -338,4 +369,132 @@ var _ = Describe("Scheduler", Ordered, func() {
 			Expect(podMap).To(HaveLen(deviceNum), "Expected exactly %d pods to be scheduled", deviceNum)
 		})
 	})
+
+	Context("topology", func() {
+		const (
+			topologyName     = "gb2000-topology"
+			rackLabelKey     = "nvidia.com/gpu.clique"
+			hostnameLabelKey = "kubernetes.io/hostname"
+
+			gpusPerNode     = 4
+			numRacks        = 2
+			numNodesPerRack = 18
+			numNodes        = numRacks * numNodesPerRack
+		)
+
+		var (
+			topologyNodes []*corev1.Node
+			topologyTree  *kueuev1alpha1.Topology
+		)
+
+		BeforeAll(func(ctx context.Context) {
+			topologyNodes = []*corev1.Node{}
+
+			nodeToRack := map[int]string{}
+			for i := range numNodes {
+				nodeToRack[i] = fmt.Sprintf("%d", i%numRacks)
+			}
+			rand.Shuffle(len(nodeToRack), func(i, j int) {
+				nodeToRack[i], nodeToRack[j] = nodeToRack[j], nodeToRack[i]
+			})
+
+			for i := 0; i < numNodes; i++ {
+				topologyNode := DefaultNodeConfig(fmt.Sprintf("t-node-%d", i))
+				topologyNode.GPUs = gpusPerNode
+				topologyNode.Labels[rackLabelKey] = fmt.Sprintf("clusterUuid.%s", nodeToRack[i])
+				topologyNode.Labels[hostnameLabelKey] = topologyNode.Name
+				nodeObj := CreateNodeObject(ctx, ctrlClient, topologyNode)
+				Expect(ctrlClient.Create(ctx, nodeObj)).To(Succeed(), "Failed to create topology test node")
+				topologyNodes = append(topologyNodes, nodeObj)
+			}
+
+			topologyTree = &kueuev1alpha1.Topology{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: topologyName,
```

Comment:
Add label `app=engine-e2e`. This helps when the tests' cleanup fails for any reason.

Reply:
You can see that we don't add this label or similar one on the other objects in the envtests

Reply:
Oh right, sorry